lib/computer_player.py

```diff
@@ -7,11 +9,74 @@ def __init__(self, board, mark):
         self.mark = mark
 
     def move(self):
-        input_number = self.next_available_space_index()
-        self.board.mark_position(input_number, self.mark)
+        time.sleep(1)
+        # index = self.next_available_space_index()
+        index = self.ai_index()
+        self.board.mark_position(index, self.mark)
         self.board.display_board()
 
     def next_available_space_index(self):
         values = list(self.board.board.values())
         input = values.index(" ")
         return input + 1
+
+    # for minimax algorithm:
+    def is_a_winner(self, board):
+        for win_comb in self.board.win_combinations:
+            if (
+                board[win_comb[0]] == board[win_comb[1]] == board[win_comb[2]]
+                and board[win_comb[0]] != " "
+            ):
+                return True
+            else:
+                return False
+
+    def get_winner(self, board, player):
+        for win_comb in self.board.win_combinations:
+            if board[win_comb[0]] == board[win_comb[1]] == board[win_comb[2]] == player:
+                return player
+
+    def is_board_full(self, board):
+        return " " not in board.values()
+
+    def get_empty_spaces(self, board):
+        return [k for k in board if board[k] == " "]
```

Comment:
This seems like logic that would better fit inside the `Board` class

Reply:
Ah, looks like you already have `get_empty_spaces` in the `Board` class, so you can remove this duplication.